lib/kafo/fact_writer.rb

```diff
@@ -0,0 +1,24 @@
+module Kafo
+  class FactWriter
+    DATA_FILENAME = 'kafo.yaml'
+    WRAPPER_FILENAME = 'kafo.rb'
+
+    def self.write_facts(facts, directory)
+      Dir.mkdir(directory)
+
+      # Write a data file containing all the facts encoded as YAML
+      File.open(File.join(directory, DATA_FILENAME), 'w') { |f| f.write(YAML.dump(facts)) }
```

Comment:
would be nice to have debug log for this (similar to elsewhere)

same few lines below

Reply:
Rather than adding logging to the FactWriter, I've done this in the ExecutionEnvironment.